### PR TITLE
Fixed CORS and added cookie based AUTH

### PR DESCRIPTION
**Pull request by Kysumi**

### CODE REVIEW

File: backend/src/server.ts [hunk 1]
```diff
@@ -35,31 +37,24 @@ createConnection()
...
-                'Origin',
-                'X-Requested-With',
-                'Content-Type',
-                'Accept',
-                'X-Access-Token'
-            ],
+        // Supporting credential calls with CORS from FETCH
+        var whitelist = ['http://localhost:3000']
+        var corsOptions: cors.CorsOptions = {
+            origin: (origin: any, callback: any) => {
```
Comment (jallier) [inline backend/src/server.ts]: Any way to type these?

Comment (jallier) [reply backend/src/server.ts]: Even something like Func

File: frontend/src/lib/fetch.ts [hunk 2]
```diff
@@ -8,9 +8,10 @@ export async function makeRequest(url: string, method = "GET", data = {}) {
   const token = localStorage.getItem("token");
   const fetchOptions: RequestInit = {
     method,
-    headers: {
-      Authorization: `Bearer ${token}`
-    }
+    // headers: {
```
Comment (jallier) [inline frontend/src/lib/fetch.ts]: We can remove this if using cookies

**Review by jallier — APPROVED** [review]

File: backend/src/server.ts (hunk 1 above)
Comment (Kysumi) [reply backend/src/server.ts]: Done

**Pull request merged**